react/containers/ChannelLoginForm/view.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { makePostRequest } from 'utils/xhr';
+import { setUserCookies } from 'utils/cookies';
+import { replaceChannelSelectionInNavBar } from 'utils/page';
+
+class ChannelLoginForm extends React.Component {
```

Comment:
Forms are one of the most painful things in React. It might be worth looking into 3rd party React form elements. I've done a fair bit of work with forms and think that Formik is really nice. Hoping to add it to more parts of the app soon. https://github.com/jaredpalmer/formik

I really like how they use the `render` prop, I think it's a really nice way to interact with React elements